proof-of-spacetime.md

```diff
@@ -112,22 +119,30 @@ Seed [32]byte
 
 ```go
 type Challenge struct {
-    Sector Uint
+    Sector Uint // an index into a list of sectors in a miner's proving set
     Leaf Uint
 }
 ```
 
+```go
+type ReplicaInfo struct {
+    Commitment [32]byte // CommR as produced by the sealing process
+    SectorID   SectorID // identity for a miner's sector
+    Path       string   // used to access the replica's contents when generating a PoSt (i.e. not required for verification)
+}
+```
+
 #### Challenge Derivation
 
 ```go
 // Derive the full set of challenges for PoSt.
-func DerivePoStChallenges(seed Seed, faults FaultSet, sectorSize Uint, sectorCount Uint) [POST_CHALLENGES_COUNT]Challenge {
+func DerivePoStChallenges(seed Seed, faults []Uint, sectorSize Uint, sectorCount Uint) [POST_CHALLENGES_COUNT]Challenge {
     challenges := []
 
     for n := 0; n < POST_CHALLENGES_COUNT; n++ {
         attemptedSectors := {SectorID:bool}
         while challenges[n] == nil {
-            challenge := DerivePoStChallenge(seed, n, faults, attempt, sectorSize, sectorCount)
+            challenge := DerivePoStChallenge(seed, n, attempt, sectorSize, sectorCount)
```

Comment:
Doesn't challenge derivation need to know the faults, in order to avoid challenging faulted sectors? Alternately, we could reduce `sectorCount` and handle indexing into the narrowed sectors outside of the challenge derivation.

I realize `DerivePoStChallenge` doesn't currently handle this, but I think it needs to be dealt with somewhere.

Reply:
> Doesn't challenge derivation need to know the faults, in order to avoid challenging faulted sectors?

The pseudocode shows that `DerivePoStChallenge` does not need to know the faults. Instead, it produces a challenge which is compared with the list of faults provided to `DerivePoStChallenges`. If the generated challenge exists in that list of faults, it is not added to the final list of challenges. Instead, a new challenge is generated and this process repeats until the required number of challenges has been generated.

Reply:
Got it — was commenting on the fly, thanks.